pl_examples/domain_templates/computer_vision_fine_tuning.py

```diff
@@ -0,0 +1,440 @@
+        self.hparams = hparams
+        self.dl_path = dl_path
+        self.__build_model()
+
+    def __build_model(self):
+        """Define model layers & loss."""
+
+        # 1. Load pre-trained network:
+        model_func = getattr(models, self.hparams.backbone)
+        backbone = model_func(pretrained=True)
+
+        _layers = list(backbone.children())[:-1]
+        self.feature_extractor = torch.nn.Sequential(*_layers)
+        freeze(module=self.feature_extractor, train_bn=self.hparams.train_bn)
+
+        # 2. Classifier:
+        _fc_layers = [torch.nn.Linear(2048, 256),
+                      torch.nn.Linear(256, 32),
+                      torch.nn.Linear(32, 1)]
+        self.fc = torch.nn.Sequential(*_fc_layers)
+
+        # 3. Loss:
+        self.loss_func = F.binary_cross_entropy_with_logits
+
+    def forward(self, x):
+        """Forward pass. Returns logits."""
+
+        # 1. Feature extraction:
+        x = self.feature_extractor(x)
+        x = x.squeeze(-1).squeeze(-1)
+
+        # 2. Classifier (returns logits):
+        x = self.fc(x)
+
+        return x
+
+    def loss(self, labels, logits):
+        return self.loss_func(input=logits, target=labels)
+
+    def train(self, mode=True):
+        super().train(mode=mode)
+
+        epoch = self.current_epoch
+        if epoch < self.hparams.milestones[0] and mode:
+            # feature extractor is frozen (except for BatchNorm layers)
+            freeze(module=self.feature_extractor,
+                   train_bn=self.hparams.train_bn)
+
+        elif self.hparams.milestones[0] <= epoch < self.hparams.milestones[1] and mode:
+            # Unfreeze last two layers of the feature extractor
+            freeze(module=self.feature_extractor,
+                   n=-2,
+                   train_bn=self.hparams.train_bn)
+
+    def on_epoch_start(self):
+        """Use `on_epoch_start` to unfreeze layers progressively."""
+        optimizer = self.trainer.optimizers[0]
+        if self.current_epoch == self.hparams.milestones[0]:
+            _unfreeze_and_add_param_group(module=self.feature_extractor[-2:],
+                                          optimizer=optimizer,
+                                          train_bn=self.hparams.train_bn)
+
+        elif self.current_epoch == self.hparams.milestones[1]:
+            _unfreeze_and_add_param_group(module=self.feature_extractor[:-2],
+                                          optimizer=optimizer,
+                                          train_bn=self.hparams.train_bn)
+
+    def training_step(self, batch, batch_idx):
+
+        # 1. Forward pass:
+        x, y = batch
+        y_logits = self.forward(x)
+        y_true = y.view((-1, 1)).type_as(x)
+        y_bin = torch.ge(y_logits, 0)
+
+        # 2. Compute loss & accuracy:
+        train_loss = self.loss(y_true, y_logits)
+        num_correct = torch.eq(y_bin.view(-1), y_true.view(-1)).sum()
+
+        # 3. Outputs:
+        tqdm_dict = {'train_loss': train_loss}
```

Comment:
no need to name it after particular progress bar lib...

Reply:
Done!